src/charm.py

```diff
@@ -116,6 +123,7 @@ def crd_resource_handler(self, handler: KubernetesResourceHandler):
 
     def _check_leader(self):
         """Check if this unit is a leader."""
+        self.framework.observe(self.on.update_status, self._on_event)
```

Comment:
Do we need this? :sweat_smile:

Reply:
Oh, haha, it must be a leftover from the rebase (it was added by mistake in another PR) :P will remove it.